echopype/convert/set_groups_azfp.py

```diff
@@ -27,14 +27,14 @@ def set_env(self) -> xr.Dataset:
         """Set the Environment group.
         """
         # TODO Look at why this cannot be encoded without the modifications -- @ngkavin: what modification?
```

Comment:
What does this line of comment mean?

Reply:
I think it means why the time can't be saved without subtracting `np.datetime64('1900-01-01T00:00:00')` and dividing by `np.timedelta64(1, 's')`.

Reply:
Have you tried? I had tried this early on, but you should confirm if that is the current behavior for xarray, in both writing to file and read from file.

Reply:
`(self.parser_obj.ping_time - np.datetime64('1970-01-01T00:00:00')) / np.timedelta64(1, 's')` converts ping_time to a float which removes the `datetime64` dtype from the ping_time. When saving a `datetime64`, ` xarray has its own attributes for 'calendar' and 'units' which we override in setGoups, but is not allowed.

Specifically, the error is:
 ` ValueError
failed to prevent overwriting existing key units in attrs on variable 'ping_time'. This is probably an encoding field used by xarray to describe how a variable is serialized. To proceed, remove this key from the variable's attributes manually.`

So the problem is only with writing and this is still what xarray 0.16.2 does as well as the latest 0.17.0.